manifests/realm.pp

```diff
@@ -115,30 +115,29 @@
 # Copyright 2013 Patrick Mooney.
 # Copyright (c) IN2P3 Computing Centre, IN2P3, CNRS
 #
-define mit_krb5::realm(
-  $kdc                    = '',
-  $master_kdc             = '',
-  $admin_server           = '',
-  $database_module        = '',
-  $default_domain         = '',
-  $v4_instance_convert    = '',
-  $v4_realm               = '',
-  $auth_to_local_names    = '',
-  $auth_to_local          = '',
-  $kpasswd_server         = '',
-  $v4_realm_convert       = '',
-  $pkinit_anchors         = '',
-  $pkinit_pool            = '',
-  Boolean $rotate_servers = false,
-  $http_anchors           = '',
+define mit_krb5::realm (
+  Optional[Array] $kdc                  = undef,
+  Optional[String] $master_kdc          = undef,
+  Optional[Array] $admin_server         = undef,
+  Optional[String] $database_module     = undef,
+  Optional[String] $default_domain      = undef,
+  Optional[String] $v4_instance_convert = undef,
+  Optional[String] $v4_realm            = undef,
+  Optional[String] $auth_to_local_names = undef,
```

Comment:
I should rework this `auth_to_local_names` logic 
This should be converted to a dedicated subsection in the `krb5.conf` file. A `Hash` type should be better, I need to think about that.